borsh/src/ser/mod.rs

```diff
@@ -201,6 +204,38 @@ impl BorshSerialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (bigint, exponent) = self.as_bigint_and_exponent();
+        bigint.serialize(writer)?;
+        exponent.serialize(writer)
+    }
+}
+
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::num_bigint::BigInt {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (sign, data) = self.to_u32_digits();
```

Comment:
Looking into their implementation, seems like https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html#method.to_bytes_le will give you the minimal amount of bytes needed (which would be the canonical form)

Reply:
@austinabell `to_bytes_le` on BigInt, indeed, returns the minimum number of bytes, but BigDecimal still needs to be canonicalized.

@telezhnaya We have to use `.normalized()`:

```rust
    let a = bigdecimal::BigDecimal::new(bigdecimal::num_bigint::BigInt::from_slice(bigdecimal::num_bigint::Sign::Plus, &[11]), 1);
    let b = bigdecimal::BigDecimal::new(bigdecimal::num_bigint::BigInt::from_slice(bigdecimal::num_bigint::Sign::Plus, &[110]), 2);
    println!("a: {} ({:?})", a, a.as_bigint_and_exponent().0.to_bytes_le());
    println!("b: {} ({:?})", b, b.as_bigint_and_exponent().0.to_bytes_le());
    println!("a == b: {:?}", a == b);
    println!("b normalized: {} ({:?})", b.normalized(), b.normalized().as_bigint_and_exponent().0.to_bytes_le());
```

Output:

```
a: 1.1 ((Plus, [11]))
b: 1.10 ((Plus, [110]))
a == b: true
b normalized: 1.1 ((Plus, [11]))
```